poly-commitment/src/lagrange_cache.rs

```diff
@@ -0,0 +1,201 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    fs::{self, File},
+    marker::PhantomData,
+    path::{Path, PathBuf},
+};
+
+use crate::{commitment::CommitmentCurve, PolyComm};
+use ark_poly::Radix2EvaluationDomain as D;
+use std::hash::{Hash, Hasher};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum LagrangeCache<G> {
+    Local(FileCache<G>),
+    Remote(GoogleCloudCache<G>),
+}
+
+impl<G> Default for LagrangeCache<G> {
+    fn default() -> Self {
+        LagrangeCache::Local(FileCache::new(PathBuf::from("/tmp/lagrange_cache")))
+    }
+}
+
+// Keep this enum private for type safety
+pub enum LagrangeCacheKey {
+    Local(PathBuf),
+    Remote(()),
+}
```

Comment:
imo a name like `LagrangeCacheFile` is a clearer name as Key implies this is for some map type. We should comment why we've called it `LagrangeCache[Key|File]` by explaining the storage scheme.

Reply:
Suggest: `source`

Reply:
This type is no longer needed, I just needed to get my rust legs back and remember how to use the languaeg